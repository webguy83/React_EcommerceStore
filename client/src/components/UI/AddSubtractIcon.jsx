import React from "react";
import styled, { css } from "styled-components/macro";

// css

const disabledStyles = css`
  opacity: 0.1;

  &:hover {
    cursor: default;
    color: inherit;
  }
`;

const AddSubtractIconContainer = styled.div`
  cursor: pointer;

  &:hover,
  &:active {
    color: var(--sec);
    transition: var(--btnTrans);
  }

  ${({ disabled }) => {
    return disabled ? disabledStyles : null;
  }}
`;

//jsx

const AddSubtractIcon = ({ addIcon, addOrSubtract, disabled }) => {
  return (
    <AddSubtractIconContainer
      data-testid="container"
      disabled={disabled}
      onClick={addOrSubtract}
    >
      <span data-testid="span" role="img" aria-label="addRemove">
        {addIcon ? <>&#10133;</> : <>&#10134;</>}
      </span>
    </AddSubtractIconContainer>
  );
};

export default AddSubtractIcon;
