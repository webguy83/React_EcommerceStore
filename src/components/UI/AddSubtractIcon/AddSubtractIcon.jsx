import React from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faMinusCircle, faPlusCircle } from '@fortawesome/free-solid-svg-icons';

import { AddSubtractIconContainer, disabledAddSubtractIcon } from './AddSubtractIcon.module.scss';

const AddSubtractIcon = ({ addIcon, addOrSubtract, disabled }) => {
    return (
        <div className={`${disabled ? disabledAddSubtractIcon : ""} ${AddSubtractIconContainer}`} onClick={addOrSubtract}>
            <FontAwesomeIcon icon={addIcon ? faPlusCircle : faMinusCircle} />
        </div>
    );
};

export default AddSubtractIcon;