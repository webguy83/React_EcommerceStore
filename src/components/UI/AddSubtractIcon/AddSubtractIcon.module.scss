@import '../../../mixins/UI.scss';  

.AddSubtractIconContainer {
    &:hover {
        @include buttonHover;
    }
}