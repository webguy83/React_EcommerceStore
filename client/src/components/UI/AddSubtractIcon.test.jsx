import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import AddSubtractIcon from "./AddSubtractIcon";
import "jest-styled-components";

afterEach(cleanup);

describe("add and subtraction icon tests", () => {
  it("renders without crashing", () => {
    const { asFragment } = render(
      <AddSubtractIcon addIcon addOrSubtract={() => {}} disabled={false} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the + icon", () => {
    const { getByTestId } = render(
      <AddSubtractIcon
        addIcon={true}
        addOrSubtract={() => {}}
        disabled={false}
      />
    );
    expect(getByTestId("span")).toHaveTextContent(/^➕$/);
  });

  it("should render the - icon", () => {
    const { getByTestId } = render(
      <AddSubtractIcon
        addIcon={false}
        addOrSubtract={() => {}}
        disabled={false}
      />
    );
    expect(getByTestId("span")).toHaveTextContent(/^➖$/);
  });

  it("should make the - icon opacity low when disabled", () => {
    const { getByTestId } = render(
      <AddSubtractIcon
        addIcon={false}
        addOrSubtract={() => {}}
        disabled={true}
      />
    );
    expect(getByTestId("container")).toHaveStyleRule("opacity", "0.1");
  });
});
