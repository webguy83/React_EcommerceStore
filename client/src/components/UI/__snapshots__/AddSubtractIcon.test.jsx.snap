// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add and subtraction icon tests renders without crashing 1`] = `
<DocumentFragment>
  .c0 {
  cursor: pointer;
}

.c0:hover,
.c0:active {
  color: var(--sec);
  -webkit-transition: var(--btnTrans);
  transition: var(--btnTrans);
}

<div
    class="c0"
    data-testid="container"
  >
    <span
      aria-label="addRemove"
      data-testid="span"
      role="img"
    >
      âž•
    </span>
  </div>
</DocumentFragment>
`;
